### PR TITLE
Mfed tests

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add test for 100% coverage on MessageFederate.cpp and make the interfaces between messageFederates and ValueFederates more consistent in C++.

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/helics/application_api/MessageFederate.cpp  2
- tests/helics/application_api/CombinationFederateTests.cpp  2
         

Clones removed
==============
+ tests/helics/application_api/MessageFederateAdditionalTests.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5147751&bid=16908526)

File: src/helics/application_api/MessageFederate.hpp [hunk 1]
```diff
@@ -33,15 +33,34 @@ class HELICS_CXX_EXPORT MessageFederate:
         const std::string& fedName,
         const std::shared_ptr<Core>& core,
         const FederateInfo& fi);
-    /**constructor taking a string with the required information
-    @param configString can be either a JSON file, TOML file or a string containing JSON code
-    */
-    explicit MessageFederate(const std::string& configString);
+
+	/**constructor taking a core and a federate information structure, sore information in fi is ignored
```
Comment (nightlark) [inline src/helics/application_api/MessageFederate.hpp]:
```suggestion
	/**constructor taking a core and a federate information structure, core information in fi is ignored
```

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/helics/application_api/CombinationFederateTests.cpp  2
- src/helics/application_api/MessageFederate.cpp  2
         

Clones removed
==============
+ tests/helics/application_api/MessageFederateAdditionalTests.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5147751&bid=16908526)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/helics/application_api/CombinationFederateTests.cpp  2
- src/helics/application_api/MessageFederate.cpp  2
         

Clones removed
==============
+ tests/helics/application_api/MessageFederateAdditionalTests.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5147751&bid=16908526)

**Pull request merged**